src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java

```diff
@@ -35,6 +42,10 @@ public BattleshipModel() {
         playerMisses= new ArrayList<>();
         computerHits = new ArrayList<>();
         computerMisses= new ArrayList<>();
+        verticalStack = new Stack<>();
+        horizontalStack = new Stack<>();
+        shotListStack = new Stack<>();
+        populateShotListStack();
```

Comment:
The code is easy to read since its Java and unlike other nasty languages such as C & C++. Your code would be made much more better by adding many more comments about the several functions you've written and also more inline comments within functions where necessary to explain the key aspects of the logic of how hard mode is being implemented in the back end.  I think the choice of variable names is informative which helps and I also feel that each function is doing 1 task and not more than that which is also nice.